src/security_buffer.rs

```diff
@@ -1,20 +1,13 @@
 use std::fmt;
 use std::mem::take;
 
-use crate::{Error, ErrorKind, Result, SecurityBufferType};
+use crate::{BufferType, Error, ErrorKind, OwnedSecurityBufferType, Result, SecurityBufferFlags};
 
-/// A special security buffer type is used for the data decryption. Basically, it's almost the same
-/// as `OwnedSecurityBuffer` but for decryption.
-///
-/// [DecryptMessage](https://learn.microsoft.com/en-us/windows/win32/secauthn/decryptmessage--general)
-/// "The encrypted message is decrypted in place, overwriting the original contents of its buffer."
+/// A security buffer type with a mutable reference to the buffer data.
 ///
-/// So, the already defined `OwnedSecurityBuffer` is not suitable for decryption because it uses [Vec] inside.
-/// We use reference in the [SecurityBuffer] structure to avoid data cloning as much as possible.
-/// Decryption/encryption input buffers can be very large. Even up to 32 KiB if we are using this crate as a TSSSP(CREDSSP)
-/// security package.
+/// Basically, it is a security buffer but without buffer flags.
 #[non_exhaustive]
-pub enum SecurityBuffer<'data> {
+pub enum SecurityBufferType<'data> {
```

Comment:
naming: What do you think of `UntaggedSecurityBuffer`? Does it make sense in this context?

Reply:
Actually, it is tagged because each enum variant represents a different buffer type. It lacks only buffer flags, so I renamed it to `UnflaggedSecurityBuffer`. What do you think?